### PR TITLE
APEXCORE-727 Added a maven module which build a binary package for apex

### PR DESCRIPTION
**Pull request by chinmaykolhatkar**
@tweise  Most of the work is used and influenced from atroto and bigtop work already been done.
The changes are tested with launching pi-demo using generated binary package on hadoop 2.7.

Please review and merge as appropriate.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -144,6 +144,7 @@
     <module>engine</module>
     <module>apex-app-archetype</module>
     <module>apex-conf-archetype</module>
+    <module>apex-bin</module>
```
Comment (tweise) [inline pom.xml]: Would `dist` be a more appropriate module name?

File: apex-bin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+                  <goal>single</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>apache-release</id>
```
Comment (tweise) [inline apex-bin/pom.xml]: This isn't required here. To the contrary, the dist build should probably only run under this profile (parent pom.xml).

File: apex-bin/pom.xml [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+            <goals>
+              <goal>copy-dependencies</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${project.build.directory}/lib</outputDirectory>
+              <includeScope>runtime</includeScope>
+            </configuration>
+          </execution>
+          <execution>
+            <id>copy-dependencies-hadoop</id>
```
Comment (tweise) [inline apex-bin/pom.xml]: Are these used in any way? If not, drop this.

File: apex-bin/apex.1 [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+
+
+.SH ENVIRONMENT
+
+.TP
+.BI JAVA_HOME
+The java implementation to use.  Overrides JAVA_HOME.
+
+.TP
+.BI DT_CLIENT_OPTS
```
Comment (tweise) [inline apex-bin/apex.1]: should be renamed to `APEX_CLIENT_OPTS`

File: apex-bin/apex.1 [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+.\" Unless required by applicable law or agreed to in writing, software
+.\" distributed under the License is distributed on an "AS IS" BASIS,
+.\" WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.\" See the License for the specific language governing permissions and
+.\" limitations under the License.
+.\"
+.\" Process this file with
+.\" groff -man -Tascii apex.1
+.\"
+.TH apex 1 "August 2015 " Linux "User Manuals"
```
Comment (tweise) [inline apex-bin/apex.1]: 2018?

File: pom.xml (hunk 1 above)
Comment (chinmaykolhatkar) [reply pom.xml]: Yes... Changed.

File: apex-bin/pom.xml (hunk 2 above)
Comment (chinmaykolhatkar) [reply apex-bin/pom.xml]: Make sense... Moved the "dist" module under modules tag of apache-release profile.

File: apex-bin/pom.xml (hunk 3 above)
Comment (chinmaykolhatkar) [reply apex-bin/pom.xml]: Nope.. Missed removing it... Removed.

File: apex-bin/apex.1 (hunk 4 above)
Comment (chinmaykolhatkar) [reply apex-bin/apex.1]: Done.

File: apex-bin/apex.1 (hunk 5 above)
Comment (chinmaykolhatkar) [reply apex-bin/apex.1]: Done.

Comment (chinmaykolhatkar) [reply apex-bin/apex.1]: That is supposed to be last update date. I'll make it May 2018

**Comment by chinmaykolhatkar** [comment]
Jenkins, test this please.

File: dist/apex.1 [hunk 6]
```diff
@@ -0,0 +1,501 @@
...
+
+
+.SH ENVIRONMENT
+
+.TP
+.BI JAVA_HOME
+The java implementation to use.  Overrides JAVA_HOME.
+
+.TP
+.BI APEX_CLIENT_OPTS
```
Comment (tweise) [inline dist/apex.1:493]: Can you please open a follow-up JIRA/PR to change the variables in the apex script (and any documentation that may refer to it)?

Comment (chinmaykolhatkar) [reply dist/apex.1:493]: Done. APEXCORE-814.

**Comment by chinmaykolhatkar** [comment]
@tweise  I'm unable to find why Jenkins maven verify build is failing... Any idea? travis seems to pass.

**Review by tweise — APPROVED** [review]

**Comment by tweise** [comment]
test this please

File: dist/apex.1 [hunk 7]
```diff
@@ -0,0 +1,501 @@
...
+Extra Java runtime options. Default is "-Xmx1024m -XX:+UseConcMarkSweepGC -XX:+CMSClassUnloadingEnabled"
+
+.TP
+.BI HADOOP_HOME
+Optionally, the Hadoop home to run with.
+
+.SH COPYRIGHT
+Copyright (C) 2015 The Apache Software Foundation. All rights reserved.
```
Comment (vrozov) [inline dist/apex.1]: Update copyright to the current year.

File: dist/pom.xml [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.apache.apex</groupId>
+    <artifactId>apex</artifactId>
+    <version>4.0.0-SNAPSHOT</version>
+  </parent>
+
+  <artifactId>apex-dist</artifactId>
+  <name>Apache Apex Binary Release Package</name>
```
Comment (vrozov) [inline dist/pom.xml]: Consider "Apache Apex Binaries Distribution package".

File: dist/apex.1 (hunk 7 above)
Comment (chinmaykolhatkar) [reply dist/apex.1]: Done

File: dist/pom.xml (hunk 8 above)
Comment (chinmaykolhatkar) [reply dist/pom.xml]: Done.

**Comment by chinmaykolhatkar** [comment]
@vrozov  @tweise  Suggested changes done.

**Pull request merged**